### ISSUE TITLE
Instant crash - error while loading shared libraries: libabsl_log_internal_check_op.so

### ISSUE THREAD

**Issue by LorenAmelang** [qgis/QGIS#55114]
### What is the bug or the crash?

qgis: error while loading shared libraries: libabsl_log_internal_check_op.so.2308.0.0: cannot open shared object file: No such file or directory

### Steps to reproduce the issue

I just installed
pacman -S -i qgis
Repository : extra
Name : qgis
Version : 3.34.0-1

It fails immediately with
qgis: error while loading shared libraries: libabsl_log_internal_check_op.so.2308.0.0: cannot open shared object file: No such file or directory

My system had:
/usr/lib/libabsl_log_internal_check_op.so.2301.0.0

I updated to:
Packages (1) abseil-cpp-20230802.1-1

Now it has:
/usr/lib/libabsl_log_internal_check_op.so.2308.0.0

But now QGIS fails with:
qgis: error while loading shared libraries: libabsl_log_internal_check_op.so.2301.0.0: cannot open shared object file: No such file or directory

### Versions

Can't get that far...

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

I can't get to editing my profile. But this was a fresh, first time install on this machine. 

This could well be an Arch problem, I've reported it there, but I'm hoping someone at QGIS will have a better understanding of why the app looks for a specific version of the library, and how it could revert to looking for the previous version once I've installed the new version it was originally looking for.

**Comment by agiudiceandrea** [comment]
Hi @LorenAmelang, it seems to me this is not an issue related to the QGIS code base, but a packaging issue probably system related. The bug report https://bugs.archlinux.org/task/80134 has been already closed by the QGIS for Arch Linux packager @antonio-rojas with the following reason: "Not a bug - Partial upgrades are not supported. Please use the support channels if you need help". Have you followed the advise?

**Comment by LorenAmelang** [comment]
A summary reply for all the places this issue was mentioned...


I began by installing the default pacman version of the QGIS package, for the first time on this machine, not an update of it or a specific version. 
It immediately failed, saying it couldn't find libabsl_log_internal_check_op.so.2308.0.0. 
Certainly seems to me like either QGIS didn't specify that as a dependency, or pacman didn't check it. 


Reply: 
If you're "installing version anything of anything" you're conducting a partial update and this is most certainly NOT a "fresh current install"

So are we now required to do a full system update whenever we install any single package? Before installing, after installing, or both? 

Aha! Looks like that's true:
https://wiki.archlinux.org/title/System_maintenance#Partial_upgrades_are_unsupported
That needs much more exposure - and should have been linked in the first reply I received to my bug report!


Reply: 
Post the output of  pacman -Qikk libabseil-cpp qgis 

It shows  error: package 'libabseil-cpp' was not found  - both before and after the full system update and restart. And still just now. 

Looks like it is: 
pacman -Qikk abseil-cpp
Name            : abseil-cpp
Version         : 20230802.1-1
Description     : Collection of C++ library code designed to augment the C++ standard library
Install Date    : Tue 31 Oct 2023 02:07:39 PM PDT
Install Reason  : Installed as a dependency for another package
abseil-cpp: 787 total files, 0 altered files

That  Tue 31 Oct 2023 02:07:39 PM PDT  date was when I tried installing the absl/abseil packages, long after I inatalled QGIS. 
So does  20230802.1-1  imply  libabsl_log_internal_check_op.so.2308.0.0  ? 

I had tried installing both python-absl-1.4.0-3 and abseil-cpp-20230802.1-1 , both appeared successful. 
The  libabsl_log_internal_check_op.so  version on disk changed. 
The QGIS error changed - after I installed what it said it wanted, it complained it couldn't find the previous version. but it still failed. 


Reply: 
It seems to me this is not an issue related to the QGIS code base, but a packaging issue probably system related. 

Seems that depends on whether QGIS specifies that  python-absl-1.4.0-3  or  abseil-cpp-20230802.1-1  are dependencies. 

Pacman only looked for:
Packages (26) assimp-5.2.5-2  blosc-1.21.5-1  draco-1.5.6-2  gdal-3.7.2-3
              geos-3.12.0-1  gsl-2.7.1-3  hdf5-1.14.2-1  libaec-1.0.6-1
              libfreexl-2.0.0-1  libgeotiff-1.7.1-2  librttopo-1.1.0-5
              libspatialite-5.1.0-1  libzip-1.10.1-1  netcdf-4.9.2-2  pdal-2.5.6-1
              proj-9.3.0-1  python-qscintilla-qt5-2.14.1-1  qhull-2020.2-4
              qscintilla-qt5-2.14.1-1  qt5-3d-5.15.11-1  qt5-location-5.15.11+kde+r5-1
              qtkeychain-qt5-0.14.1-2  qwt-6.2.0-1  spatialindex-1.9.3-1
              xerces-c-3.2.4-3  qgis-3.34.0-1

Maybe one of those should have included it? Far beyond my knowledge...  


At least I've fumbled my way to a working QGIS...
